server/bin/pbench-state-manager.py

```diff
@@ -50,14 +51,37 @@ def main(options):
         Database.init_db(config, logger)
 
         args = {}
-        if options.user:
-            args["owner"] = options.user
+        if options.create:
+            user = options.create
+            try:
+                user = Auth.validate_user(user)
+            except UnknownUser:
+                # FIXME: I don't want to be creating the user here or
+                # dealing with a non-existing user. The unittest setup
+                # should create the test users we want ahead of time
+                # using a pbench-user-manager command and we should
+                # depend on having them here! The following is a hack
+                # until that command exists.
+                #
+                # The desired behavior would be to remove this try and
+                # except and allow UnknownUser to be handled below with
+                # an error message and termination.
```

Comment:
Normally I would want the `UnknownUser` to terminate with a message. Until I have a separate utility to create users, I wanted to create the `User` here, which implies catching the exception and then acting. There'd be no point in a `try` block "below" as we don't want to handle other exceptions (e.g., if something goes wrong with creating the `User` or `Dataset`) because we want them to terminate with a message.

Maybe I'm not sure what you're getting at?

Reply:
It's pretty trivial.  I would propose the same code, but different sequencing.  I wouldn't put the temporary creation of the `User` embedded in the argument handling code; rather, I would place it later in the function, where it can be set off better from the surrounding structures and more easily nuked at the appropriate time in the future.

There's no reason why what you have isn't workable, it's just that the same construct is embedded a few levels deeper and it interrupts the readability of the surrounding structure.  But, on the assumption that this will all go away in a few weeks, it probably doesn't matter.

Reply:
I'm inclined to keep this as it is; if I deferred the `User` creation I'd have to remember that the validation (which I do want to be part of the "argument handling") failed so I can use that later, which would be a broader effect on the logic.